### PR TITLE
Move Steps view before Deploy

### PR DESCRIPTION
**Pull request by johnsca**
Addresses #1009

The inputs for all of the post-deploy steps will be collected prior to the deploy screen, so that once a deployment is started, all of the interactive bits will be done and the user can just walk away.

### CODE REVIEW

**Comment by johnsca** [comment]
From Travis:

```
Traceback (most recent call last):
  File "/opt/python/3.5.3/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/home/travis/build/conjure-up/conjure-up/conjureup/controllers/runsteps/tui.py", line 12, in run_steps
    for step in app.steps:
TypeError: 'NoneType' object is not iterable
```

**Comment by johnsca** [comment]
Steps should be getting loaded [here](https://github.com/conjure-up/conjure-up/pull/1037/files#diff-ec3ed04dc18a4bc15b58f828912325b9R7).  There must be some flow that I haven't found that causes that view to be skipped.

File: conjureup/app.py [hunk 1]
```diff
@@ -233,6 +228,15 @@ def main():
     app.state = redis.StrictRedis(host='localhost',
                                   port=opts.redis_port,
                                   db=0)
+    # confirm that the Redis connection is working
+    # XXX perhaps we should start the service if not running
```
Comment (battlemidget) [inline conjureup/app.py]: Yea since we dont require root this could be easily handled in conjure-up

File: conjureup/app_config.py [hunk 2]
```diff
@@ -122,10 +122,19 @@ class AppConfig:
     state = None
 
     # Sentry endpoint
-    sentry = None,
+    sentry = None
```
Comment (battlemidget) [inline conjureup/app_config.py]: ugh nice catch that was my mistake still thinking it was a simplenamespace

File: conjureup/app.py (hunk 1 above)
Comment (battlemidget) [inline conjureup/app.py]: https://github.com/conjure-up/conjure-up/issues/1039

**Pull request merged**